### ISSUE TITLE
CleanFilesReporter: comparison of Hash with Hash failed

### ISSUE THREAD

**Issue by AlfredoRamos** [brigade/scss-lint#930]
I get this error when using the `--format=CleanFiles` option:

```shell
$ scss-lint --format=CleanFiles .
comparison of Hash with Hash failed
/home/alfredo/.gem/ruby/2.4.0/gems/scss_lint-0.56.0/lib/scss_lint/reporter/clean_files_reporter.rb:7:in `sort'                                                                                          
/home/alfredo/.gem/ruby/2.4.0/gems/scss_lint-0.56.0/lib/scss_lint/reporter/clean_files_reporter.rb:7:in `report_lints'                                                                                  
/home/alfredo/.gem/ruby/2.4.0/gems/scss_lint-0.56.0/lib/scss_lint/cli.rb:193:in `block in report_lints'                                                                                                 
/home/alfredo/.gem/ruby/2.4.0/gems/scss_lint-0.56.0/lib/scss_lint/cli.rb:192:in `each'              
/home/alfredo/.gem/ruby/2.4.0/gems/scss_lint-0.56.0/lib/scss_lint/cli.rb:192:in `report_lints'      
/home/alfredo/.gem/ruby/2.4.0/gems/scss_lint-0.56.0/lib/scss_lint/cli.rb:68:in `scan_for_lints'     
/home/alfredo/.gem/ruby/2.4.0/gems/scss_lint-0.56.0/lib/scss_lint/cli.rb:59:in `act_on_options'     
/home/alfredo/.gem/ruby/2.4.0/gems/scss_lint-0.56.0/lib/scss_lint/cli.rb:33:in `run'                
/home/alfredo/.gem/ruby/2.4.0/gems/scss_lint-0.56.0/bin/scss-lint:7:in `<top (required)>'           
/home/alfredo/.gem/ruby/2.4.0/bin/scss-lint:23:in `load'                                            
/home/alfredo/.gem/ruby/2.4.0/bin/scss-lint:23:in `<main>'
```
SCSS-Lint version: 0.56.0
Sass version: 3.5.3
Ruby version: 2.4.2
.scss-lint.yml
```yaml
scss_files:
  - 'scss/**/*.scss'

linters:
  BorderZero:
    enabled: true
    convention: none
  Comment:
    enabled: true
    exclude:
      - scss/main.scss
  ChainedClasses:
    enabled: true
  Indentation:
    enabled: true
    character: tab
    width: 1
  PropertySortOrder:
    enabled: true
    order: concentric
```

**Comment by AlfredoRamos** [comment]
Thanks @srawlins

For me `clean_files` return an array of hashes `[{path: 'file'}]`, and got it working with the following change:

```diff
diff --git a/lib/scss_lint/reporter/clean_files_reporter.rb b/lib/scss_lint/reporter/clean_files_reporter.rb
index 4f7ba54..7470cfb 100644
--- a/lib/scss_lint/reporter/clean_files_reporter.rb
+++ b/lib/scss_lint/reporter/clean_files_reporter.rb
@@ -3,7 +3,7 @@ module SCSSLint
   class Reporter::CleanFilesReporter < Reporter
     def report_lints
       dirty_files = lints.map(&:filename).uniq
-      clean_files = files - dirty_files
+      clean_files = (files - dirty_files).map { |file| file[:path] }
       clean_files.sort.join("\n") + "\n" if clean_files.any?
     end
   end
```

However with this change tests fail. Surely there's a better and proper way to fix it.

**Comment by srawlins** [comment]
Well, I guess you should be... proud to find a 2 year old bug! This was introduced in 8ad337a1a9d38f890b56ac355e44aadd36b7aa80, in Nov 2015 😄 . Your patch is the right patch; I've opened #938.

**Comment by AlfredoRamos** [comment]
Installing the gem from GitHub seems to fix the problem but it does not print anything, I'll need to check if that's a desired output.

Thanks.
